internal/store/store.go

```diff
@@ -487,3 +500,49 @@ func (s *Store) GetImage(composeID uuid.UUID) (*Image, error) {
 
 	return nil, &NotFoundError{"compose could not be found"}
 }
+
+func (s *Store) PushSource(source SourceConfig) {
+	s.change(func() error {
+		s.Sources[source.Name] = source
+		return nil
+	})
+}
+
+func (s *Store) DeleteSource(name string) {
+	s.change(func() error {
+		delete(s.Sources, name)
+		return nil
+	})
+}
+
+func (s *Store) ListSources() []string {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	names := make([]string, 0, len(s.Sources))
+	for name := range s.Sources {
+		names = append(names, name)
+	}
+	sort.Strings(names)
+
+	return names
+}
+
+func (s *Store) GetSource(name string, source *SourceConfig) bool {
```

Comment:
Hm, why doesn't this just return (a copy of) the SourceConfig or `nil`?

Reply:
I based this function off of the GetBlueprint function which sets the blueprint through a pointer and returns a bool if it is found. Would you recommend I change this to a copy so we don't pass a reference to the store?

Reply:
Yeah, I had preferred that. But let's hear with @larskarlitski (assuming he did the original?), he might see something that I miss.